bonfhir/bonfhir
# Add `dayjs` and other missing peer dependencies to Next and Vite templates

Fixes #265 

The `@mantine/dates` library defines `dayjs >=1.0.0` as a peer dependency and recommends installing the `dayjs` dependencies in its [README][0].

I believe that the _getting started_ value of the `@mantine/dates` library is worth keeping it in the template so its requirements should be met when the template is generated.

The same needed to be done for peer dependencies of `@bonfhir/tiptap`, `@mantine/tiptap`, `@tiptap/react`.

## How can I test this?

I ran the following from the top-level directory of the Bonfhir repo on this branch:

```shell
pnpm run build
cd packages/cli
node ./dist/cli/index.cjs create
```

From there, select the Vite or Next templates, select yarn as a package manager, the run `yarn dev` in the project directory. This should no longer yield any peer dependency-related errors.

## Note

I don't think accounting for peer dependencies in templates will be too burdensome at this point but if it ever becomes a hassle to maintain, we might want to consider having some kind of template generation test that validates peer dependencies. We could add this CLI test to a daily CI job that would warn us if a template is no longer in a _good state_ when created today. I'm not convinced it's worth the time investment for now though.

[0]: https://github.com/mantinedev/mantine/tree/master/packages/%40mantine/dates
